src/meta-srv/src/lease.rs

```diff
@@ -270,6 +295,12 @@ impl PeerLookupService for MetaPeerLookupService {
             .map_err(BoxedError::new)
             .context(common_meta::error::ExternalSnafu)
     }
+    async fn frontends(&self) -> common_meta::error::Result<Vec<Peer>> {
+        lookup_frontends(&self.meta_peer_client, u64::MAX)
+            .await
+            .map_err(BoxedError::new)
+            .context(common_meta::error::ExternalSnafu)
+    }
```

Comment:
How about adding a flag to optionally return the active peer addresses, using `lookup_frontends(&self.meta_peer_client, u64::MAX, FRONTEND_HEARTBEAT_INTERVAL_MILLIS / 2000)`?